src/robot/libraries/Collections.py

```diff
@@ -698,60 +748,108 @@ def get_from_dictionary(self, dictionary, key, default=NOT_SET):
+        as an argument
         """
         self._validate_dictionary(dictionary)
-        self.dictionary_should_contain_key(dictionary, key, msg)
+        should_ignore = self._is_ignore_case(ignore_case)
+        self.dictionary_should_contain_key(dictionary, key, msg, should_ignore['key'])
+        if ignore_case is True or self._normalize(ignore_case)=="both":
+            dictionary = self._normalize(dictionary)
+            key=self._normalize(key)
+            value=self._normalize(value)
+        elif should_ignore['key']:
+            dictionary = {self._normalize(k): dictionary[k] for k in dictionary}
+            key=self._normalize(key)
+        elif should_ignore['value']:
+            dictionary = {k: self._normalize(dictionary[k]) for k in dictionary}
+            value=self._normalize(value)
         assert_equal(dictionary[key], value,
                      msg or f"Value of dictionary key '{key}' does not match",
                      values=not msg)
 
-    def dictionary_should_contain_value(self, dictionary, value, msg=None):
+    def dictionary_should_contain_value(self, dictionary, value, msg=None,
+                                        ignore_case=False):
         """Fails if ``value`` is not found from ``dictionary``.
 
         Use the ``msg`` argument to override the default error message.
+        
+        See section `Ignore Case` for more information about how to use ignore_case 
+        as an argument
         """
         self._validate_dictionary(dictionary)
+        should_ignore = self._is_ignore_case(ignore_case)
+        if should_ignore['value']:
+            dictionary=self._normalize(dictionary)
+            value=self._normalize(value)
         _verify_condition(value in dictionary.values(),
                           f"Dictionary does not contain value '{value}'.",
                           msg)
 
-    def dictionary_should_not_contain_value(self, dictionary, value, msg=None):
+    def dictionary_should_not_contain_value(self, dictionary, value, msg=None,
+                                            ignore_case=False):
         """Fails if ``value`` is found from ``dictionary``.
 
         Use the ``msg`` argument to override the default error message.
+        
+        See section `Ignore Case` for more information about how to use ignore_case 
+        as an argument
         """
         self._validate_dictionary(dictionary)
+        should_ignore = self._is_ignore_case(ignore_case)
+        if should_ignore['value']:
+            dictionary = self._normalize(dictionary)
+            value=self._normalize(value)
         _verify_condition(value not in dictionary.values(),
                           f"Dictionary contains value '{value}'.",
                           msg)
 
-    def dictionaries_should_be_equal(self, dict1, dict2, msg=None, values=True):
+    def dictionaries_should_be_equal(self, dict1, dict2, msg=None, values=True,
+                                     ignore_case=False):
```

Comment:
This keyword has got new `ignore_keys` argument since this PR was created. It creates a conflict, but it ought to be pretty easy to resolve.

Responsibility to actually ignore keys could be moved to the `Normalize` class. Probably not a good idea to do that as part of this PR, though.